hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -335,7 +361,10 @@ public boolean rename(final Path src, final Path dst) throws IOException {
     }
 
     // Non-HNS account need to check dst status on driver side.
-    if (!abfsStore.getIsNamespaceEnabled() && dstFileStatus == null) {
+    TracingContext tracingContext = new TracingContext(clientCorrelationID,
+        fileSystemID, HdfsOperationConstants.RENAME, true, tracingContextFormat,
+        listener);
+    if (!abfsStore.getIsNamespaceEnabled(tracingContext) && dstFileStatus == null) {
```

Comment:
refactored the getFileStatus methods in AzureBlobFileSystem class to accommodate these. Changes done:

- overload getFileStatus to use context passed by other methods
- tryGetFileStatus to take context as arg
- pvt method createFileSystem signature change to avoid overloading tryGetFileStatus